z1randomizer.py

```diff
@@ -22,6 +22,14 @@
     'resetting door repair charges.',
 )
 
+# TODO: Turn this enum into a string with lambda validation.
+flags.DEFINE_enum(
+    'level_text',
+    'level-',
+    ['level-', 'house-', 'block-', 'random', 'cage_-', 'home_-', 'castle'],
```

Comment:
I could add a check that prevents re-writing the same value as the original, though I admit I am not sure I see the benefit. Perhaps I'm misunderstanding something from your explanation?

Reply:
The check has been placed within `text_data_table` now. If you feel this is good, you can resolve this portion of the conversation.

If Github doesn't let you for whatever reason, I can do it.

Reply:
Works for me.  I think the future way to fix this would be to have a mapping between the value of the flag from the user's perspective and the actual string -- that'd also better support the idea of a randomly chosen string.  But that can be future work -- no need to worry about it in this change.